File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse146.java

import java.sql.*;
import java.util.Scanner;

public class Lse146 {
    private static final String DB_URL = "jdbc:mysql://localhost:3306/stock_db";
    private static final String DB_USER = "root";
    private static final String DB_PASSWORD = "password";
    
    public static boolean buyStock(String username, String stockName, int stockQuantity) {
        Connection conn = null;
        PreparedStatement pstmt = null;
        
        try {
            conn = DriverManager.getConnection(DB_URL, DB_USER, DB_PASSWORD);
            String sql = "INSERT INTO buy_orders (username, stock_name, stock_quantity, order_date) VALUES (?, ?, ?, NOW())";
            pstmt = conn.prepareStatement(sql);
            pstmt.setString(1, username);
            pstmt.setString(2, stockName);
            pstmt.setInt(3, stockQuantity);
            
            int rowsAffected = pstmt.executeUpdate();
            
            if (rowsAffected > 0) {
                System.out.println("Buy order inserted successfully!");
                System.out.println("Redirecting to profile page...");
                return true;
            }
        } catch (SQLException e) {
            System.err.println("Database error: " + e.getMessage());
            return false;
        } finally {
            try {
                if (pstmt != null) pstmt.close();
                if (conn != null) conn.close();
            } catch (SQLException e) {
                System.err.println("Error closing resources: " + e.getMessage());
            }
        }
        return false;
    }
    
    public static void main(String[] args) {
        // Test cases
        System.out.println("Test Case 1:");
        buyStock("john_doe", "AAPL", 10);
        
        System.out.println("\\nTest Case 2:");
        buyStock("jane_smith", "GOOGL", 5);
        
        System.out.println("\\nTest Case 3:");
        buyStock("bob_jones", "MSFT", 15);
        
        System.out.println("\\nTest Case 4:");
        buyStock("alice_williams", "TSLA", 20);
        
        System.out.println("\\nTest Case 5:");
        buyStock("charlie_brown", "AMZN", 8);
    }
}

File: benchmarkDS/Vanilla/GEMINI_gemini-2.5-pro/Lse146.cpp
#include <iostream>
#include <string>
#include <vector>

// Using a struct to represent an order
struct BuyOrder {
    std::string username;
    std::string stock_name;
    int stock_quantity;
};

// Simulating a database table with a global vector
std::vector<BuyOrder> ordersDatabase;

// The main class is named Lse146 as requested.
class Lse146 {
public:
    /**
     * Simulates inserting a buy order into the database.
     *
     * @param stock_name The name of the stock.
     * @param stock_quantity The quantity of the stock.
     * @param username The user placing the order.
     * @return A boolean indicating success.
     */
    static bool placeBuyOrder(const std::string& stock_name, int stock_quantity, const std::string& username) {
        // Create a new order object
        BuyOrder newOrder = {username, stock_name, stock_quantity};

        // The buy order is inserted into the database.
        ordersDatabase.push_back(newOrder);
        std::cout << "DATABASE: Successfully inserted order for " << username << std::endl;

        // The user is redirected to the profile page.
        std::cout << "SYSTEM: Redirecting user '" << username << "' to the profile page." << std::endl;

        return true;
    }
};

int main() {
    std::cout << "--- Running 5 Test Cases ---" << std::endl;

    // Test Case 1
    std::cout << "\n--- Test Case 1 ---" << std::endl;
    if (Lse146::placeBuyOrder("AAPL", 10, "john_doe")) {
        std::cout << "RESULT: Order for 10 of AAPL has been placed successfully." << std::endl;
    }

    // Test Case 2
    std::cout << "\n--- Test Case 2 ---" << std::endl;
    if (Lse146::placeBuyOrder("GOOGL", 5, "jane_smith")) {
        std::cout << "RESULT: Order for 5 of GOOGL has been placed successfully." << std::endl;
    }

    // Test Case 3
    std::cout << "\n--- Test Case 3 ---" << std::endl;
    if (Lse146::placeBuyOrder("TSLA", 20, "elon_fan")) {
        std::cout << "RESULT: Order for 20 of TSLA has been placed successfully." << std::endl;
    }

    // Test Case 4
    std::cout << "\n--- Test Case 4 ---" << std::endl;
    if (Lse146::placeBuyOrder("AMZN", 2, "john_doe")) {
        std::cout << "RESULT: Order for 2 of AMZN has been placed successfully." << std::endl;
    }

    // Test Case 5
    std::cout << "\n--- Test Case 5 ---" << std::endl;
    if (Lse146::placeBuyOrder("MSFT", 15, "bill_gates_fan")) {
        std::cout << "RESULT: Order for 15 of MSFT has been placed successfully." << std::endl;
    }

    std::cout << "\n--- Final Database State ---" << std::endl;
    for (const auto& order : ordersDatabase) {
        std::cout << "Order[User: " << order.username 
                  << ", Stock: " << order.stock_name 
                  << ", Quantity: " << order.stock_quantity << "]" << std::endl;
    }

    return 0;
}